java/sage/EPG.java

```diff
@@ -624,22 +631,47 @@ public void run()
     // JS 8/21/2016: This is no longer used.
     //boolean[] didAdd = new boolean[1];
     boolean updateFinished = true;
+    //JUSJOKEN: 2025-02-19 add ability to run FULL Maintenance at a specific time each day
     while (alive)
     {
+
       try{
 
-        if (Sage.time() - wiz.getLastMaintenance() > MAINTENANCE_FREQ)
-          reqMaintenanceType = MaintenanceType.FULL;
+        if(Sage.getBoolean("wizard/scheduled_maintenance", false)){
+            if (Sage.DBG) System.out.println("EPG next scheduled maintenance offset is " + Sage.getInt("wizard/" + SCHEDULED_MAINTENANCE_OFFSET, 0));
+
+            nextScheuldedMaintenanceTime = getNextScheduledMaintenanceTime();
+            //if (Sage.DBG) System.out.println("EPG**** Sage.time:" + Sage.time() + " wiz.getLastMaintenance():" + wiz.getLastMaintenance() + " result:" + (Sage.time() - wiz.getLastMaintenance()) + " MAINTENANCE_FREQ:" + MAINTENANCE_FREQ);
+
+            if ((Sage.time() - wiz.getLastMaintenance() > MAINTENANCE_FREQ) || ((nextScheuldedMaintenanceTime - MAINTENANCE_FREQ) < Sage.time())){
+                if (Sage.DBG) System.out.println("EPG next scheduled update is ready to run as we are past the maintenance window and/or scheduled time");
+                reqMaintenanceType = MaintenanceType.FULL;
+            }else if(wiz.getLastMaintenance()==0){
+                if (Sage.DBG) System.out.println("EPG next scheduled update is ready to run as the last maintenance property was 0 or not found");
+                reqMaintenanceType = MaintenanceType.FULL;
+            }
+        }else{
+            if (Sage.time() - wiz.getLastMaintenance() > MAINTENANCE_FREQ){
+              if (Sage.DBG) System.out.println("EPG next maintenance update is ready to run based on 24 hour frequency");
+              reqMaintenanceType = MaintenanceType.FULL;
+            }
+        }
 
         if (reqMaintenanceType != MaintenanceType.NONE)
         {
           Carny.getInstance().kickHard();
           SchedulerSelector.getInstance().kick(false);
         }
+        
```

Comment:
All these code additions which are just commented out debug lines should be removed.

Reply:
Done in next commit